### PR TITLE
FP-2926: Fixed a couple of console errors

### PR DESCRIPTION
**Pull request by manuelnogueiramov**
Addresses [FP-2926](https://movai.atlassian.net/browse/FP-2926)

* Added a script to validate typescript
* Made document reading more resilient by showing a console.error instead of not catching the error and having that invasive backdrop with the error.
image
* Made reading the configuration more resilient by showing a console.error if we can't read the configuration for some reason.
* Used optional chaining to prevent the Robot not Online error.
image

[FP-2926]: https://movai.atlassian.net/browse/FP-2926?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by diasnad — DISMISSED** [review]

File: src/api/RobotManager/RobotManager.ts [hunk 1]
```diff
@@ -251,9 +251,9 @@ class RobotManager {
     clearTimeout(this.heartbeatTimeout);
     const robotsWhichChangedOnlineStatus = [];
     Object.values(this.robots).forEach(robot => {
-      const previousOnlineStatus = this.cachedRobots[robot.id].Online;
+      const previousOnlineStatus = this.cachedRobots[robot.id]?.Online;
```
Comment (diasdm) [inline src/api/RobotManager/RobotManager.ts:254]: Would the changes in https://github.com/MOV-AI/frontend-npm-lib-core/pull/194/files make more sense?

File: src/api/Document/DocumentV1.js [hunk 2]
```diff
@@ -58,15 +58,20 @@ class DocumentV1 {
   /**
    * Get document data
    */
-  read = (apiVersion = "v1") => {
+  read = async (apiVersion = "v1") => {
```
[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/api/Document/DocumentV1.js:61]: It was always returning a promise `return Rest.get({ path }).then(data => {`

File: src/api/RobotManager/RobotManager.ts (hunk 1 above)
Comment (manuelnogueiramov) [reply src/api/RobotManager/RobotManager.ts:254]: My changes allow `checkStatus(robot)` to run anyway and check the status of the actual robot regardless of, if it was cached or not. The changes on that PR just straight do nothing if the robot isn't cached,

File: src/api/Document/DocumentV1.js [hunk 3]
```diff
@@ -58,15 +58,20 @@ class DocumentV1 {
...
-      return apiVersion === "v1" ? data : data[type][name];
-    });
+    try {
+      const documentData = await Rest.get({ path });
+
+      this.data = documentData;
+
+      return apiVersion === "v1" ? documentData : documentData[type][name];
+    } catch (e) {
+      console.log(`Error reading document "${name}" of type "${type}"`, e);
```
[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/api/Document/DocumentV1.js:73]: Nice catch. Fixed.

[the remark this answers is not in the document]
Comment (diasdm) [reply src/api/Document/DocumentV1.js:73]: No push :eyes:

[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/api/Document/DocumentV1.js]: I was checking if there were any more changes before pushing this, and kinda forgot to push afterwards... :facepalm:

[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/api/Document/DocumentV1.js]: It's done now.

**Review by diasdm — APPROVED** [review]

**Pull request merged**